### ISSUE TITLE
Add CI

### ISSUE THREAD

**Issue by JonathanLorimer** [JonathanLorimer/htmx#9]

**Comment by srid** [comment]
Hey @JonathanLorimer - I'd be happy to contribute a PR to enable CI, but the flake is currently not building:

```
error: Package ‘2captcha-0.1.0.0’ in /nix/store/y8pbnccb8bc1p8fchx7zkb0874yblrg3-source/pkgs/development/haskell-modules/hackage-packages.nix:12 is marked as broken, refusing to evaluate.
```

BTW, I'd be willing to switch to using [haskell-flake](https://github.com/srid/haskell-flake) in this flake (and fix the above issue in the process) if that is something you'd find desirable.

PS: I came across your package [through this reddit post](https://www.reddit.com/r/haskell/comments/1egnq8e/ann_htmx0001_a_library_for_using_htmx_in_haskell/) since I'm looking to use htmlx in a new project of mine.

**Comment by JonathanLorimer** [comment]
@srid can you elaborate on how its not building? The flake is building for me.

**Comment by srid** [comment]
You're right, my issue was due to #15 - after resolving which I can build the whole flake using [`om ci`](https://omnix.page/om/ci.html).

**Comment by srid** [comment]
One more thing: the use of `pkgs.haskell.packages.${compilerVersion}` means that Haskell packages will be compiled on the local machine; whereas if we use the default compiler's package set (`pkgs.haskellPackage`) they will be fetched from the Nix binary cache. This means that CI can finish in minutes rather than hours, and we don't need to use a binary cache.

![image](https://github.com/user-attachments/assets/d96d4544-9db7-4436-bab8-f5dc25d97683)

With this patch, it should be straightforward to add a GitHub actions workflow to build the flake in CI.

**Comment by JonathanLorimer** [comment]
@srid Sorry it has taken me so long to get to this. I am going to mess around with the compiler set locally myself. I think it would be nice to have a matrix of compilers we build / test against. Perhaps using the default compiler for the default package is the best.

I am a bit wary of `om ci`, I'd rather use something a bit more generic such as github actions, in order to make things more approachable.
